src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsMSBuildProjectSystem.cs

```diff
@@ -727,6 +727,19 @@ public VSLangProj.References References
             }
         }
 
+        public VSLangProj157.References3 References3
+        {
+            get
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+
+                dynamic projectObj = VsProjectAdapter.Project.Object;
+                var references = (VSLangProj157.References3)projectObj.References;
```

Comment:
should we differentiate between NPE vs other exceptions? like if it fails to type cast into `Reference3`

Reply:
Hmm. What if we reported a telemetry event if the type cast fails? That should only happen if there's a setup authoring issue (e.g. the type library wasn't registered correctly or something like that) or we're encountering a project system that implements `References` but not `References3`. Either way we would want to know.

Does that sound reasonable?

Reply:
yeah that sounds good.

Reply:
@jainaashish I've added a very simple telemetry event.